Network/Socket/Types.hsc

```diff
@@ -941,7 +941,10 @@ type CSaFamily = (#type sa_family_t)
 -- in that the value of the argument /is/ used.
 sizeOfSockAddr :: SockAddr -> Int
 #if defined(DOMAIN_SOCKET_SUPPORT)
-sizeOfSockAddr SockAddrUnix{}  = #const sizeof(struct sockaddr_un)
+sizeOfSockAddr (SockAddrUnix path) =
+    case path of
+        '\0':_ -> (#const sizeof(sa_family_t)) + length path
```

Comment:
The abstract sockets are a Linux-specific feature.  Should support for this also be Linux-specific?  Is there any evidence of support for such unix-domain sockets on any of the BSDs, or other unix-like system?

[ By the way, it is rather unfortunate that the `path` associated with a unix-domain socket `SockAddrUnix` is a `String` and not a `ByteString`. Calling `length` here is only correct, because when binding the socket we later truncate all the `Char`s to `CChar`s.  Using `ByteString` here would have been a better interface. ]

Reply:
Also at around line 1019, I don't see any effort to decode abstract socket addresses correctly.  Is there a portable way to distinguish empty paths from abstract socket addresses?  Presently it looks like they'll return an empty path from `accept()` , `getsockname()` and `getpeername()`.  If these are to be supported, surely not just `connect()`, `sendto()` and `bind()` should work, but also `accept()`, `getsockname()`, `getpeername` and `recvfrom()` for datagram sockets.